curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-tone-of-a-color.md

```diff
@@ -20,9 +20,10 @@ All elements have a default `background-color` of `transparent`. Our `nav` eleme
 The `nav` element should have a `background-color` of the adjusted cyan tone using the `hsl()` property.
 
 ```js
-assert(
-  code.match(/nav\s*?{\s*?background-color:\s*?hsl\(180,\s*?80%,\s*?25%\)/gi)
-);
+assert.equal(
+  new __helpers.CSSHelp(document).getStyle('nav').getPropVal('background-color', true), 
+  'rgb(13,115,115)'
```

Comment:
It would be better to use `toEqual` with the documented target hsl values, so it's clear that the test is correct.

Reply:
@lynxlynxlynx I am not sure I understand your request correctly. Just to be clear you're asking to change test to:
```js
expect (
  new __helpers.CSSHelp(document).getStyle('nav').getPropVal('background-color', true)
).toEqual('rgb(13,115,115)')
```

Reply:
Kinda, but with the hsl syntax the test is about.

Reply:
From the comments in #45859 I am left with the impression that the point of this change is to test against computed style value as opposed to `hsl` value i.e. using regex.
Regarding the use of BDD syntax (e.g. `expect().toEqual()`) I have 2 concerns.
First, I haven't tried yet but I am not sure the syntax is available in the challenge tests and second if it does work, this test will be an exception from all others. I checked and this pattern is not used in any of the other tests.

Reply:
The main idea is to use a robust test instead of regex matching (rarely robust). You're making the test harder to understand by hardcoding some value, instead of letting the browser do the computation from something that's obviously correct.

Reply:
Can you give an example of how to get the `hsl` value instead of hard coding it?

Reply:
It's unavoidable to hardcode something in this case, I'm just saying you should pick the clearer choice — the thing mentioned by the challenge instructions.

Reply:
This is were I struggle to understand your request for change. Can you post version of this assertion you think is more suitable?

Reply:
```js
expect(new __helpers.CSSHelp(document).getStyle('nav').getPropVal('background-color', true)).toEqual('hsl(180, 80%, 25%)');
```

Reply:
I can't do that because 
```
new __helpers.CSSHelp(document).getStyle('nav').getPropVal('background-color', true)
```
evaluates the style and returns the `rgb` value, hence `'rgb(13,115,115)'`.

Reply:
I have also tried using the `expect().toEqual()` syntax but it seems it's not supported. With the requested changes this test will never pass when it should.

Reply:
What about creating in test some dummy element, setting `background-color` for it using expected `hsl` values, and then comparing `background-color` of the dummy element with `background-color` of the tested element?

Reply:
Oh, sorry, I thought that evaluates, not comparing plainly. I'm then fine with just an extra comment that the hsl values resolve to the used rgb ones.

Reply:
It might be possible but from a single line test it will grow into something far more complex and difficult to understand.
I also don't see how the test value could cause confusion. The test is not visible to the user. If you think it might be confusing for maintainers I can add comment and/or add explanation in the relevant commit message.

Reply:
> What about creating in test some dummy element, setting `background-color` for it using expected `hsl` values

It might work but from a one liner the test will become unnecessarily complicated IMO.
I don't see how it could be confusing for the user since it is not visible and as for the maintainers the comment I have just added should explain the use of the `rbg` values.